### DISCUSSION TITLE
Vaultwarden update discrepancy - CLI startup log shows 1.32.4 but Admin panel shows 1.25.2

### DISCUSSION THREAD

**Discussion by bonjitzer** [dani-garcia/vaultwarden#5177] [General]
Hello, I am wondering if I do have the latest vaultwarden update based on the discrepancy mentioned in the title.

I did the commands listed in https://github.com/dani-garcia/vaultwarden/wiki/Updating-the-vaultwarden-image to update my image via Docker.

![Screenshot 2024-11-11 124219](https://github.com/user-attachments/assets/1ab7bfd7-e339-4727-b4ab-d8fefe8e15ca)
![vaultwarden_logs](https://github.com/user-attachments/assets/305b31ea-7af7-4eb4-94d7-b8f6e8746858)

**Comment by BlackDex** [comment]
The server installed should mention the correct and same version as the console does. If not, sometime strange is happening.

**Reply by bonjitzer** [reply]
Any ideas to troubleshoot? Bitwarden app on iOS stopped working, so I saw that I should update and no matter what method I've tried to update with results in the admin panel showing an older version installed.

**Reply by BlackDex** [reply]
Check `docker ps`, i suspect you are running more then 1 Vaultwarden.

Also, try to switch to docker compose. That is a bit easier to manage

**Reply by bonjitzer** [reply]
docker ps only showed the 1 vaultwarden container:
![docker ps](https://github.com/user-attachments/assets/90186a28-bc9c-4f6b-b35d-61a85d887d7e)

switched over to docker-compose with the following config:
![docker-compose-file](https://github.com/user-attachments/assets/a874b78c-2db6-4886-87ea-4f3d69e7274c)

Still seeing the following:
![docker-compose-ps](https://github.com/user-attachments/assets/2f35e5eb-40aa-4555-a9df-88d169105662)
![docker-compose-admin](https://github.com/user-attachments/assets/6473fcee-d561-41bb-9504-2c4b74db3998)
![docker-compose-cli](https://github.com/user-attachments/assets/29144692-ae3f-42a2-8570-510ed275d89d)

**Reply by BlackDex** [reply]
did you run the following?
```bash
docker compose pull
docker compose up -d
```

That should have re-created the container with the new version.
If not, then something on your system is causing issues.

Also be sure to use the latest docker version v20.19+ if I'm correct. But docker is already v27.10.
You can check this by running `docker info`

**Reply by bonjitzer** [reply]
Removed all vaultwarden images, updated docker to latest, pulled latest image, compose up -d and still same results.
![newest_docker-compose](https://github.com/user-attachments/assets/9e2ec6ce-d697-4aa1-8f6e-67c551110bf0)

Any other thoughts? And thank you for your assistance thus far.

**Reply by kpfleming** [reply]
It would appear that your Docker installation is not pulling images from Docker Hub, but from somewhere else which does not have the latest images available.

**Reply by BlackDex** [reply]
Browser or reverse proxy cache?
I really have no clue. If the CLI/Console output shows the right version, it should be running the right version.
The only other thing i can think of is that you are connecting to the wrong server, but that would be strange.

Try using a different browser or InCognito/Private mode.

**Reply by bonjitzer** [reply]
> It would appear that your Docker installation is not pulling images from Docker Hub, but from somewhere else which does not have the latest images available.

As far as I can tell, I have not added any other registry than default. Portainer also shows Docker Hub (Anonymous) as the only registry. I have pulled images both through Portainer as well as via docker pull in the CLI.

**Reply by bonjitzer** [reply]
> Browser or reverse proxy cache? I really have no clue. If the CLI/Console output shows the right version, it should be running the right version. The only other thing i can think of is that you are connecting to the wrong server, but that would be strange.
> 
> Try using a different browser or InCognito/Private mode.

Same behavior in an incognito window. I attempted to access vaultwarden directly via IP to bypass reverse proxy, but vaultwarden requires HTTPS. I'll look into clearing cache from my reverse proxy and see.

**Reply by bonjitzer** [reply]
Tried the testing branch and still no change in the admin panel, but it did show the testing branch on the console:
![vaultwarden_docker-compose_testing_branch](https://github.com/user-attachments/assets/302fe45b-611e-4976-9c41-dd27c9130845)
![vaultwarden_console_testing_branch](https://github.com/user-attachments/assets/c22d8e63-27e2-4a8d-921e-d53d46951f9b)
![vaultwarden_diagnostics_testing_branch](https://github.com/user-attachments/assets/d229ead8-bc41-46d1-9e24-6712d8ceea92)


I wonder if there is something in the /data directory that is not being updated or permissions issue not allowing so... I'm gonna try a fresh install using the latest in a new directory and see.

Support String:
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.25.2
* Web-vault version: v2022.6.2
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: false (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.*********.*****",
  "domain_origin": "*****://*********.*********.*****",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": null,
  "smtp_from": "****.*******@*****.***",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****.*****.***",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "****.*******@*****.***",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Reply by BlackDex** [reply]
The data directory has nothing to do with it. The version is built-in into the code. You should trace everything on how everything is connected. We can't really help with that. 

I would suggest starting with stopping Vaultwarden and see if you can still access it for some strange reason.

**Reply by bonjitzer** [reply]
> The data directory has nothing to do with it. The version is built-in into the code. You should trace everything on how everything is connected. We can't really help with that.
> 
> I would suggest starting with stopping Vaultwarden and see if you can still access it for some strange reason.

Sure enough, I had another instance of vaultwarden running on another server and my reverse proxy was pointed to that instead of the one I've been working on.... very embarrassing. 

Thank you so much for your time, patience, and insight! 
![image](https://github.com/user-attachments/assets/ea0c5f6c-2387-4a4e-a111-680ddce10103)
